### DISCUSSION TITLE
Using features from the dataset

### DISCUSSION THREAD

**Discussion by lindeberg25** [MilesCranmer/PySR#442] [Q&A]
Hello,

I've just started using the PySR tool. Here's my challenge: I'm working with a dataset that consists of 79 features and a binary label (1 or 0).  This is a simple binary classification problem. I'm interested in exploring whether PySR can help me extract a mathematical expression from this dataset. I've tried some of the provided examples, but I haven't been able to use all 79 features/variables (or less) in the final mathematical expression. 

Thanks in advance

**Comment by MilesCranmer** [comment]
Hi @lindeberg25,

Can you post your dataset and what you’ve tried and the current results? Be sure to read the tuning page for general tips: https://astroautomata.com/PySR/tuning/

79 features is quite a lot. To use all of them you’ll have to use at least `maxsize=160`, but probably twice that much to use them effectively. I’ve never gone that high so I’m not sure how it will work.

Cheers,
Miles

**Reply by lindeberg25** [reply]
Hi @MilesCranmer,

Thank you for the quick reply. Here is my dataset:

![dataset](https://github.com/MilesCranmer/PySR/assets/19540271/89d5f8df-c3c3-4405-ad2c-3223d91d6cfa)

This dataset represents the model's output derived from a model based on BERT (I have already selected the best features from the original dataset, which was much larger). I use this data to train a Random Forest for binary classification. However, I am investigating whether it is possible to understand how these features interact to make predictions. With a large number of variables, I am not sure how interpretable a mathematical equation would be.

To be honest, I am not sure if symbolic regression is well-suited for this problem. If not, I am open to other suggestions

Thank you
